diff -up nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php.omv~ nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php
--- nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php.omv~	2026-01-16 14:06:55.000000000 +0100
+++ nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php	2025-12-31 09:24:29.000000000 +0100
@@ -174,7 +174,7 @@ class Container implements \ArrayAccess
             throw new ExpectedInvokableException('Service definition is not a Closure or invokable object.');
         }
 
-        $this->factories->attach($callable);
+        $this->factories->offsetSet($callable);
 
         return $callable;
     }
diff -up nextcloud/3rdparty/pimple/pimple/src/Pimple/Exception/ExpectedInvokableException.php.omv~ nextcloud/3rdparty/pimple/pimple/src/Pimple/Exception/ExpectedInvokableException.php
diff -up nextcloud/3rdparty/pimple/pimple/src/Pimple/Exception/FrozenServiceException.php.omv~ nextcloud/3rdparty/pimple/pimple/src/Pimple/Exception/FrozenServiceException.php
diff -up nextcloud/3rdparty/pimple/pimple/src/Pimple/Exception/InvalidServiceIdentifierException.php.omv~ nextcloud/3rdparty/pimple/pimple/src/Pimple/Exception/InvalidServiceIdentifierException.php
diff -up nextcloud/3rdparty/pimple/pimple/src/Pimple/Exception/UnknownIdentifierException.php.omv~ nextcloud/3rdparty/pimple/pimple/src/Pimple/Exception/UnknownIdentifierException.php
diff -up nextcloud/3rdparty/pimple/pimple/src/Pimple/Psr11/Container.php.omv~ nextcloud/3rdparty/pimple/pimple/src/Pimple/Psr11/Container.php
diff -up nextcloud/3rdparty/pimple/pimple/src/Pimple/Psr11/ServiceLocator.php.omv~ nextcloud/3rdparty/pimple/pimple/src/Pimple/Psr11/ServiceLocator.php
diff -up nextcloud/3rdparty/pimple/pimple/src/Pimple/ServiceIterator.php.omv~ nextcloud/3rdparty/pimple/pimple/src/Pimple/ServiceIterator.php
diff -up nextcloud/3rdparty/pimple/pimple/src/Pimple/ServiceProviderInterface.php.omv~ nextcloud/3rdparty/pimple/pimple/src/Pimple/ServiceProviderInterface.php
diff -up nextcloud/lib/versioncheck.php.omv~ nextcloud/lib/versioncheck.php
--- nextcloud/lib/versioncheck.php.omv~	2026-01-22 02:33:19.606727281 +0100
+++ nextcloud/lib/versioncheck.php	2026-01-22 02:33:52.429999209 +0100
@@ -13,10 +13,10 @@ if (PHP_VERSION_ID < 80100) {
 	exit(1);
 }
 
-// Show warning if >= PHP 8.5 is used as Nextcloud is not compatible with >= PHP 8.5 for now
-if (PHP_VERSION_ID >= 80500) {
+// Show warning if >= PHP 9.0 is used as Nextcloud is not compatible with >= PHP 9.0 for now
+if (PHP_VERSION_ID >= 90000) {
 	http_response_code(500);
-	echo 'This version of Nextcloud is not compatible with PHP>=8.5.<br/>';
+	echo 'This version of Nextcloud is not compatible with PHP>=9.0.<br/>';
 	echo 'You are currently running ' . PHP_VERSION . '.';
 	exit(1);
 }
